crates/turbo-tasks-fs/src/lib.rs

```diff
@@ -200,6 +204,20 @@ impl DiskFileSystem {
         Ok(())
     }
 
+    /// registers the path as an invalidator for the current task,
+    /// has to be called within a turbo-tasks function. It removes and returns
+    /// the current list of invalidators.
+    fn register_sole_invalidator(&self, path: &Path) -> Result<HashSet<Invalidator>> {
```

Comment:
Could this be merged with `register_invalidator`. The difference seems to be whether it returns the old value, which could just be discarded by the callers of the old `register_invalidator`.

Reply:
I don't see how we get two different behaviors out of these two pieces of code:

```rust
# inserts a new invalidator
self.invalidator_map.insert(path_to_key(path), invalidator);

# replaces old invalidators
let mut invalidator_map = self.invalidator_map.lock().unwrap();
let old_invalidators = invalidator_map.insert(path_to_key(path), [invalidator].into());
```

They're both calling `map.insert(key)`, so how does one append and one replaces?

Reply:
These are different `insert` methods. `InvalidatorMap::insert` adds a single invalidator to the list.

`HashMap::insert` replaces the current list with a different list.